pkg/provider/provider.go

```diff
@@ -101,22 +110,59 @@ func DSN(s *schema.ResourceData) (string, error) {
 		region = ""
 	}
 
-	dsn, err := gosnowflake.DSN(&gosnowflake.Config{
-		Account:  account,
-		User:     username,
-		Region:   region,
-		Password: password,
-		Role:     role,
-	})
+	var dsn string
+	var err error
+
+	if len(pathPrivateKey) != 0 {
+
+		privateKeyFilename := "/Users/anneku/.ssh/snowflake_key"
+
+		// reading the private key
+		privateKeyBytes, err := ioutil.ReadFile(privateKeyFilename)
+		if err != nil || len(privateKeyBytes) == 0 { // both conditionals had to be false in order to get past this
```

Comment:
I would actually split these into two separate contidionals that way each can get a more specific error message. 
Also, go just released a feature to "wrap" error messages. so for errors, instead of %s you can do %w (to **w**rap)